src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp

```diff
@@ -745,14 +748,102 @@ namespace OpenLoco::World::MapGenerator
         });
     }
 
+    static void placeBuildingsAlongLine(World::Pos2 origin, const BuildingObject* buildingObj, const size_t buildingId, int16_t remainingDistance, uint8_t rotation)
+    {
+        // We'll be placing two buildings along the line, working towards the origin.
```

Comment:
why only 2 the old code did it until `remainingDistance -= buildingObj->averageNumberOnMap` was less than 0.